cli/pkg/cmd/search/search.go

```diff
@@ -0,0 +1,167 @@
+	formatter "github.com/tektoncd/hub/cli/pkg/formatter"
+	"github.com/tektoncd/hub/cli/pkg/printer"
+)
+
+const resTemplate = `
+{{- $rl := len .Resources }}{{ if eq $rl 0 -}}
+No Resources found
+{{ else -}}
+NAME	KIND	DESCRIPTION	TAGS
+{{- range $_, $r := .Resources -}}
+	{{- formatName $r.Name $r.LatestVersion.Version }} \
+	{{ $r.Kind }}	\
+	{{ formatDesc $r.LatestVersion.Description }} \
+	{{ formatTags $r.Tags }}
+{{- end -}}
+{{- end -}}
+`
+
+var (
+	funcMap = template.FuncMap{
+		"formatName": formatter.FormatName,
+		"formatDesc": formatter.FormatDesc,
+		"formatTags": formatter.FormatTags,
+	}
+	tmpl = template.Must(template.New("List Resources").Funcs(funcMap).Parse(resTemplate))
+)
+
+//type GlobalOpts {
+//HubURL string
+//}
+
+type options struct {
+	//*globals.GlobalOpts
+
+	cli    app.CLI
+	Limit  int
+	Match  string
+	Output string
+	Tags   []string
+	Kinds  []string
+	Args   []string
+}
+
+func Command(cli app.CLI) *cobra.Command {
+
+	opts := &options{cli: cli}
+
+	cmd := &cobra.Command{
+		Use:          "search",
+		Short:        "Search resource by combination of its name, kind, and tags",
+		Long:         ``,
+		SilenceUsage: true,
+		Args:         cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Args = args
+			return opts.run()
+		},
+	}
+
+	//opts.GlobalOpts.Init(cmd)
+
+	cmd.Flags().IntVarP(&opts.Limit, "limit", "l", 0, "Max number of resources to fetch")
+	cmd.Flags().StringVar(&opts.Match, "match", "contains", "Accept type of search. 'exact' or 'contains'.")
+	cmd.Flags().StringArrayVar(&opts.Kinds, "kinds", nil, "Accepts a comma separated list of kinds")
+	cmd.Flags().StringArrayVar(&opts.Tags, "tags", nil, "Accepts a comma separated list of tags")
+	cmd.Flags().StringVarP(&opts.Output, "output", "o", "table", "Accepts output format: [table, json]")
+
+	return cmd
+}
+
+func inList(option, val string, list []string) error {
+	val = strings.ToLower(val)
+
+	for _, v := range list {
+		if v == val {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid value %q  set for option %s. Valid options: [ %s ]",
+		val, option, strings.Join(list, ","))
+}
+
+func (opts *options) validate() error {
+	if err := inList("match", opts.Match, []string{"contains", "exact"}); err != nil {
+		return err
+	}
+
+	if err := inList("output", opts.Output, []string{"table", "json"}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (opts *options) name() string {
+	if len(opts.Args) == 0 {
+		return ""
+	}
+	return strings.TrimSpace(opts.Args[0])
+}
+
+func (opts *options) kinds() []string {
+	return opts.Kinds
+}
+
+func (opts *options) run() error {
+
+	if err := opts.validate(); err != nil {
+		return err
+	}
+
+	hub := opts.cli.Hub()
+
+	result := hub.Search(app.SearchOption{
+		Kinds: opts.kinds(),
+		Name:  opts.name(),
+	})
+
+	out := opts.cli.Stream().Out
+	if opts.Output == "json" {
+		return printer.New(out).JSON(result.Raw())
+	}
+
+	return opts.print(result.Typed())
```

Comment:
Seeing that we already get `opts.cli.Stream().Out` in this function **and** `opts.print` only uses that from `opts), I think this could be something more like

```go
type, err := result.Typed()
// handle error
return printer.New(out).Tabbed(typed, tmpl)
```